src/charts/bar.js

```diff
@@ -164,7 +166,16 @@ define(function(require) {
          * @param {BarChartData} _data The data to attach and generate the chart
          */
         function exports(_selection) {
-            _selection.each(function(_data) {
+            if (locale && enableLabels) {
+                setDefaultLocale(locale)
+                    .then(() => initChart(_selection))
+            } else {
+                initChart(_selection);
+            }
+        }
+
+        function initChart(selection) {
+            selection.each(function(_data) {
```

Comment:
I would say to trigger the locale request here and keep a reference to the promise. Something like:
```
  localeRequest = fetchLocale();
  selection.each(....
```
Later on, whenever the locale is actually needed, we'll do:
```
  localeRequest.then(...
```

That way we'll trigger the request as soon as possible (if needed) while blocking the render of anything that depends on it. 

We should probably include a configurable timeout in case the request fails.

Reply:
Liked your suggestion, i've implemented it like this.